src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,52 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final double MAX_VALUE = 100;
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private Figure figure;
+
+    public Figure getRandomFigure() {
+        int index = random.nextInt(FigureType.values().length);
+        double randomDouble = random.nextDouble() * MAX_VALUE;
+        switch (FigureType.values()[index]) {
+            case CIRCLE:
+                figure = new Circle(FigureType.CIRCLE,
+                        randomDouble,
+                        colorSupplier.getRandomColor());
+                break;
```

Comment:
```suggestion
                return new Circle(FigureType.CIRCLE,
                        randomDouble,
                        colorSupplier.getRandomColor());
```
так само можна зробити з рештою блоків